src/components/bixicard.js

```diff
@@ -35,13 +42,14 @@ export default class BixiCard extends React.Component {
             return res.json();
         })
         .then(res => {
+            let stationArray = [];
             res.stations.filter(function(station){
-                // milton/clark station 6209
-                return station.n === "6209";
+                return stationList.indexOf(station.n) > -1;
             }).forEach(function(station){
-                // set state of bikes available
-                scope.setState({bikesAvailable: station.ba});
+                stationArray.push(station);
             });
+            scope.setState({stationData: stationArray});
+            this.cycleBixi();
```

Comment:
The render() method will be retriggered as a result of setState(). Wondering whether cycleBixi() is redundant for this reason or maybe to think of whether it depends on the value that was provided to setState. Treat this as if cycleBixi() could happen before setState() completed.